pandas/_libs/window/aggregations.pyx

```diff
@@ -307,6 +308,14 @@ cdef inline void add_var(float64_t val, float64_t *nobs, float64_t *mean_x,
         return
 
     nobs[0] = nobs[0] + 1
+
+    # GH#42064, record num of same values to remove floating point artifacts
+    if val == prev_value[0] and val != MAXfloat64 and val != MINfloat64:
```

Comment:
why do max/min float matter here?

Reply:
It seems infs will be converted to nans in `_prep_values` already: https://github.com/pandas-dev/pandas/blob/afec0e9c1b2856cf28071327141a0ac0e4ba8aee/pandas/core/window/rolling.py#L354-L377
Then this inf check will be unnecessary